sdk/cloudmachine/Azure.CloudMachine/src/AIFoundary/AzureAIProjectsExensions.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        EvaluationsClient evaluationsClient = workspace.Subclients.Get(() =>
+        {
+            AIProjectClient aiClient = workspace.Subclients.Get(() => CreateAzureAIClient(workspace));
+            return aiClient.GetEvaluationsClient();
+        });
+
+        return evaluationsClient;
+    }
+
+    private static AIProjectClient CreateAzureAIClient(this ClientWorkspace workspace)
+    {
+        ClientConnection connection = workspace.GetConnectionOptions(typeof(AIProjectClient).FullName);
+        var connectionString = connection.Locator;
+        return new AIProjectClient(connectionString, workspace.Credential);
+    }
+#endregion AIProjects
+
+#region Inference
+    /// <summary>
+    /// Gets the chat completion client.
+    /// </summary>
+    /// <param name="workspace"></param>
+    /// <returns></returns>
+    public static ChatCompletionsClient GetChatCompletionsClient(this ClientWorkspace workspace)
+    {
+        ChatCompletionsClient chatClient = workspace.Subclients.Get(() => CreateChatCompletionsClient(workspace));
+        return chatClient;
+    }
+
+    private static ChatCompletionsClient CreateChatCompletionsClient(this ClientWorkspace workspace)
+    {
+        ClientConnection connection = workspace.GetConnectionOptions(typeof(ChatCompletionsClient).FullName);
+        return new(connection.ToUri(), new AzureKeyCredential(connection.ApiKeyCredential!));
+    }
+
+    /// <summary>
+    /// Gets the embeddings client.
+    /// </summary>
+    /// <param name="workspace"></param>
+    /// <returns></returns>
+    public static EmbeddingsClient GetEmbeddingsClient(this ClientWorkspace workspace)
+    {
+        EmbeddingsClient embeddingsClient =  workspace.Subclients.Get(() => CreateEmbeddingsClient(workspace));
+        return embeddingsClient;
+    }
+
+    private static EmbeddingsClient CreateEmbeddingsClient(this ClientWorkspace workspace)
+    {
+        ClientConnection connection = workspace.GetConnectionOptions(typeof(ChatCompletionsClient).FullName);
+        return new(connection.ToUri(), new AzureKeyCredential(connection.ApiKeyCredential!));
+    }
+#endregion Inference
+
+#region Azure AI Search
+    /// <summary>
+    /// Gets the search client.
+    /// </summary>
+    /// <param name="workspace"></param>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    public static SearchClient GetSearchClient(this ClientWorkspace workspace, string indexName)
+    {
+        SearchClient searchClient = workspace.Subclients.Get(() => CreateSearchClient(workspace, indexName));
+        return searchClient;
+    }
+
+    private static SearchClient CreateSearchClient(this ClientWorkspace workspace, string indexName)
+    {
+        ClientConnection connection = workspace.GetConnectionOptions(typeof(SearchClient).FullName);
+        return new(connection.ToUri(), "indexName", new AzureKeyCredential(connection.ApiKeyCredential!));
```

Comment:
Assuming you meant to pass the value here rather than the literal string?
```suggestion
        return new(connection.ToUri(), indexName, new AzureKeyCredential(connection.ApiKeyCredential!));
```

Reply:
Thank you for catching this issue!